Task5Test/src/test/java/com/company/TriangleTest.java

```diff
@@ -1,38 +1,95 @@
 package com.company;
 
 import org.testng.Assert;
-import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.DataProvider;
 import org.testng.annotations.Test;
 
+import java.math.BigDecimal;
+
 public class TriangleTest {
 
     Triangle triangle;
 
-    @BeforeMethod
-    public void setUp() throws Exception {
-        triangle = new Triangle(2, 2, 4);
-        triangle.definesType();
+    @DataProvider(name = "ValidIsoscelesTriangle")
+    public Object[][] setValidSidesOfTheIsoscelesTriangle() {
+        return new Object[][] {
+                {BigDecimal.valueOf(10), BigDecimal.valueOf(10), BigDecimal.valueOf(11)},
+                {BigDecimal.valueOf(10), BigDecimal.valueOf(11), BigDecimal.valueOf(10)},
+                {BigDecimal.valueOf(11), BigDecimal.valueOf(10), BigDecimal.valueOf(10)},
+
+                {BigDecimal.valueOf(Double.MAX_VALUE), BigDecimal.valueOf(Double.MAX_VALUE), BigDecimal.valueOf(11)},

```

Comment:
I don't think there is a sense to test Double.MAX_VALUE if you use BigDecimal. All such values will be in different equivalence classes for the double arguments, not BigDecimal.